packages/langium/src/utils/cst-util.ts

```diff
@@ -57,29 +56,51 @@ export function toDocumentSegment(node: CstNode): DocumentSegment {
     };
 }
 
-export function findRelevantNode(cstNode: CstNode): AstNode | undefined {
-    let n: CstNode | undefined = cstNode;
-    do {
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        const element = n.element as any;
-        if (element.$type !== DatatypeSymbol) {
-            return element;
+/**
+ * Performs `findLeafNodeAtOffset` with a minor difference: When encountering a non word character (i.e. not a letter, number, etc.),
+ * it will instead return the leaf node at the `offset - 1` position.
+ *
+ * For LSP services, users expect that the declaration of an element is available if the cursor is directly after the element.
+ */
+export function findDeclarationNodeAtOffset(cstNode: CstNode | undefined, offset: number): LeafCstNode | undefined {
+    if (cstNode) {
+        const localOffset = offset - cstNode.offset;
+        const textAtOffset = cstNode.text.substring(localOffset, localOffset + 1);
+        // The \p{L} regex matches any unicode letter character, i.e. characters from non-english alphabets
+        // Together with \w it matches any kind of character which can commonly appear in IDs
+        if (/^[^\w\p{L}]$/u.test(textAtOffset)) {
```

Comment:
How do we handle languages with name patterns containing non-word characters, like `kebab-case` identifiers? Do they need to override all services using this function?

Reply:
I agree, this would be limited to rare cases such as `function(identifier-|)` (with `|` being the cursor position), but that's still highly language dependent. Instead I will refactor this function to operate on a terminal+keyword condition.

Reply:
I've made the function a bit better configurable. Users can now override the `GrammarConfig` DI object to control the behavior of the regexp.